inst/templates/ad_advs.R

```diff
@@ -0,0 +1,552 @@
+  NA_character_, "WTAPCTL", "Weight-for-age percentile", 6,
+  NA_character_, "BMISDS", "BMI-for-age z-score", 7,
+  NA_character_, "BMIPCTL", "BMI-for-age percentile", 8,
+  NA_character_, "HDCSDS", "HDC-for-age z-score", 9,
+  NA_character_, "HDCPCTL", "HDC-for-age percentile", 10,
+  NA_character_, "WGTHSDS", "Weight-for-length/height Z-Score", 11,
+  NA_character_, "WGTHPCTL", "Weight-for-length/height Percentile", 12,
+  NA_character_, "BMIHSDS", "BMI-for-length/height Z-Score", 13,
+  NA_character_, "BMIHPCTL", "BMI-for-length/height Percentile", 14,
+  NA_character_, "HDCHSDS", "HDC-for-length/height Z-Score", 15,
+  NA_character_, "HDCHPCTL", "HDC-for-length/height Percentile", 16
+)
+attr(param_lookup$VSTESTCD, "label") <- "Vital Signs Test Short Name"
+
+# Get list of ADSL vars required for derivations
+adsl_vars <- exprs(SEX, BRTHDTC, TRTSDT, TRTEDT, TRT01A, TRT01P)
+
+advs <- vs %>%
+  # Join ADSL with VS (need BRTHDT for AAGECUR derivation)
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = exprs(!!!get_admiral_option("subject_keys"))
+  ) %>%
+  mutate(BRTHDT = convert_dtc_to_dt(
+    BRTHDTC,
+    highest_imputation = "n",
+    date_imputation = "first",
+    min_dates = NULL,
+    max_dates = NULL,
+    preserve = FALSE
+  )) %>%
+  ## Calculate ADT, ADY ----
+  derive_vars_dt(
+    new_vars_prefix = "A",
+    dtc = VSDTC
+  ) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT)) %>%
+  ## Calculate Current Analysis Age AAGECUR ----
+  derive_vars_duration(
+    new_var = AAGECUR,
+    new_var_unit = AAGECURU,
+    start_date = BRTHDT,
+    end_date = ADT,
+    out_unit = "DAYS",
+    trunc_out = FALSE
+  )
+
+advs <- advs %>%
+  ## Add PARAMCD only - add PARAM etc later ----
+  derive_vars_merged_lookup(
+    dataset_add = param_lookup %>% filter(!is.na(VSTESTCD)),
+    new_vars = exprs(PARAMCD),
+    by_vars = exprs(VSTESTCD)
+  ) %>%
+  ## Calculate AVAL ----
+  mutate(AVAL = VSSTRESN)
+
+## Get visit info ----
+# See also the "Visit and Period Variables" vignette
+# (https://pharmaverse.github.io/admiral/articles/visits_periods.html#visits)
+advs <- advs %>%
+  # Derive Timing
+  mutate(
+    ATPTN = VSTPTNUM,
+    ATPT = VSTPT,
+    AVISIT = case_when(
+      str_detect(VISIT, "UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
+      !is.na(VISIT) ~ str_to_title(VISIT),
+      TRUE ~ NA_character_
+    ),
+    AVISITN = as.numeric(case_when(
+      VISIT == "SCREENING 1" ~ "-1",
+      VISIT == "BASELINE" ~ "0",
+      str_detect(VISIT, "WEEK") ~ str_trim(str_replace(VISIT, "WEEK", "")),
+      TRUE ~ NA_character_
+    ))
+  )
+
+# Add Current HEIGHT Temporary variable (in cm)
+## Calculate Current HEIGHT/LENGTH at each time point Temporary variable----
+hgttmp <- advs %>%
+  filter((PARAMCD == "HEIGHT" | PARAMCD == "LENGTH") & VSSTRESU == "cm") %>%
+  select(STUDYID, USUBJID, HGTTMP = AVAL, HGTTMPU = VSSTRESU, AVISIT)
```

Comment:
there's cases in this template of STUDYID, USUBJID hard-coded, but we should use `get_admiral_option("subject_keys")` so this is flexible and can be changed from the default by the user in one place. see this admiral core PR for example for the updates needed: https://github.com/pharmaverse/admiral/pull/2452/files